model/map/src/main/java/org/keycloak/models/map/events/MapEventStoreProvider.java

```diff
@@ -0,0 +1,186 @@
+
+        MapAuthEventEntity entity = modelToEntity(event);
+        String realmId = event.getRealmId();
+        if (realmId != null) {
+            RealmModel realm = session.realms().getRealm(realmId);
+            if (realm != null && realm.getEventsExpiration() > 0) {
+                entity.setExpiration(
+                        TimeAdapter.fromIntegerWithTimeInSecondsToLongWithTimeAsInSeconds(Time.currentTime())
+                                + realm.getEventsExpiration());
+            }
+        }
+
+        authEventsTX.create(entity);
+    }
+
+    private boolean filterExpired(ExpirableEntity event) {
+        Long expiration = event.getExpiration();
+        // Check if entity is expired
+        if (expiration != null && expiration < TimeAdapter.fromIntegerWithTimeInSecondsToLongWithTimeAsInSeconds(Time.currentTime())) {
+            // Remove entity
+            authEventsTX.delete(event.getId());
+
+            return false; // Do not include entity in the resulting stream
+        }
+
+        return true; // Entity is not expired
+    }
+
+    @Override
+    public EventQuery createQuery() {
+        LOG.tracef("createQuery()%s", getShortStackTrace());
+        return new MapAuthEventQuery(((Function<QueryParameters<Event>, Stream<MapAuthEventEntity>>) authEventsTX::read)
+                .andThen(s -> s.filter(this::filterExpired).map(EventUtils::entityToModel)));
+    }
+
+    @Override
+    public void clear() {
+        LOG.tracef("clear()%s", getShortStackTrace());
+        authEventsTX.delete(QueryParameters.withCriteria(DefaultModelCriteria.criteria()));
+    }
+
+    @Override
+    public void clear(RealmModel realm) {
+        LOG.tracef("clear(%s)%s", realm, getShortStackTrace());
+        authEventsTX.delete(QueryParameters.withCriteria(DefaultModelCriteria.<Event>criteria()
+                .compare(Event.SearchableFields.REALM_ID, ModelCriteriaBuilder.Operator.EQ, realm.getId())));
+    }
+
+    @Override
+    public void clear(RealmModel realm, long olderThan) {
+        LOG.tracef("clear(%s, %d)%s", realm, olderThan, getShortStackTrace());
+        authEventsTX.delete(QueryParameters.withCriteria(DefaultModelCriteria.<Event>criteria()
+                .compare(Event.SearchableFields.REALM_ID, ModelCriteriaBuilder.Operator.EQ, realm.getId())
+                .compare(Event.SearchableFields.TIME, ModelCriteriaBuilder.Operator.LT, olderThan)
+        ));
+    }
+
+    @Override
+    public void clearExpiredEvents() {
+        LOG.tracef("clearExpiredEvents()%s", getShortStackTrace());
+
+        authEventsTX.delete(QueryParameters.withCriteria(DefaultModelCriteria.<Event>criteria()
+                        .compare(Event.SearchableFields.EXPIRATION, ModelCriteriaBuilder.Operator.LE,
+                                TimeAdapter.fromIntegerWithTimeInSecondsToLongWithTimeAsInSeconds(Time.currentTime()))));
+
+        adminEventsTX.delete(QueryParameters.withCriteria(DefaultModelCriteria.<AdminEvent>criteria()
```

Comment:
Not sure if this should be here or if we should have a separate method from admin events. In future, we may remove this method completely so I left this here. WDYT?